src/extra/AutoUI/utils.ts

```diff
@@ -1,5 +1,34 @@
 import { AutoUIBaseResource, Permissions } from './schemaOps';
 import { TFunction } from '../../hooks/useTranslation';
+import { createBrowserHistory } from 'history';
+
+export const history = createBrowserHistory();
+
+export const findInObject = (obj: any, key: string): any => {
+	let result;
+	for (const property in obj) {
+		if (obj.hasOwnProperty(property)) {
+			if (property === key) {
+				return obj[key]; // returns the value
+			} else if (typeof obj[property] === 'object') {
+				// in case it is an object
+				result = findInObject(obj[property], key);
+
+				if (typeof result !== 'undefined') {
+					return result;
+				}
+			}
+		}
+	}
+};
```

Comment:
I think we may be taking extra steps here by checking if the object has the property before checking if the property matches the key. If the property matches the key then we can just return the value, right? And *if* it does not match the key we can check if `obj[property]` is of type object (if the property does not exist on the object this check will not return true so we do not need the `hasOwnProperty` check).

```suggestion
export const findInObject = (obj: any, key: string): any => {
	for (const property in obj) {
	  if (property === key) {
	    return obj[key]; // same as obj[property]
	  } else if (typeof obj[property] === 'object') {
	    const result = findInObject(obj[property], key);
	    
	    if (typeof result !== 'undefined') {
			  return result;
		  }
	  }
	}
};
```

Reply:
@myarmolinsky the check is done since this is how to iterate over the enumerable properties of an object without executing on inherited properties. Also declaring the `result` inside the `for`  would redeclare the same variable N times and this is something you don't want for reasons of space and performance.